### ISSUE TITLE
Telescope Incompatibity

### ISSUE THREAD

**Issue by xyrintech** [GeneaLabs/laravel-model-caching#226]
**Describe the bug**
ErrorException  : Undefined offset: 0

  at /home/ubuntu/workspace/new/vendor/laravel/telescope/src/Watchers/ModelWatcher.php:36

**Environment**
 - PHP: 7.3
 - OS: Ubuntu 16.x
 - Laravel: 5.8
 - Model Caching: 0.4.11

**Comment by mikebronner** [comment]
@xyrintech That seems to be a Laravel Telescope issue. Please provide a full stack trace so we can further investigate.

**Comment by ggomuk** [comment]
Hi 

I have the same problem. finally found a reason.

```
Illuminate\Foundation\Bootstrap\HandleExceptions::handleError("Undefined offset: 0", "/var/www/vendor/laravel/telescope/src/Watchers/ModelWatcher.php", ["eloquent.updated: App\AC\AcUser"])
```

In Traits/Cacheable
use Fico7489\Laravel\Pivot\Traits\PivotEventTrait > ExtendFireModelEventTrait

That Override Model fireModelEvent so 

```php
$payload = ['model' => $this, 'relation' => $relationName, 'pivotIds' => $ids, 'pivotIdsAttributes' => $idsAttributes];

return !empty($result) ? $result : static::$dispatcher->{$method}("eloquent.{$event}: ".static::class, $payload);
```


In Laravel Telescope Model Watch 

```php
$model = FormatModel::given($data[0]);
$changes = $data[0]->getChanges();
```

That’s reason.

**Comment by corsair** [comment]
Also experiencing the same. Seems disabling ModelWatch via the Telescope config may be the best option for the time being.

Seems identical to https://github.com/GeneaLabs/laravel-model-caching/issues/185 though.

**Comment by mikebronner** [comment]
@ggomuk Thanks for tracking this down. This issue is logged here:
https://github.com/fico7489/laravel-pivot/issues/55

**Comment by mikebronner** [comment]
Reopening this, as I am having issues running Telescope as well. Will need to do some further investigation to see where this comes from.

**Comment by mikebronner** [comment]
For now I added a work-around in the README and included the Model Watcher class in the code-base to make it easy to implement for Laravel Model Caching users. Looking forward to seeing Laravel Pivot package being updated to be compatible with Telescope.

**Comment by mikebronner** [comment]
@xyrintech @ggomuk @corsair Please give release 0.4.19 or above a shot and let me know how it works for you. I updated the README with instructions for Telescope compatibility. It's just a work-around for now.

**Comment by mikebronner** [comment]
Re-opening this, as the Telescope front-end will not display custom watchers.

**Comment by mikebronner** [comment]
@xyrintech @ggomuk @corsair  This has now been fixed as of version 0.4.21 and will no longer require a work-around. It should work with Telescope out of the box. Please let me know if you are still experiencing an issue with this after updating. Thanks!
